lib/phonegap/cordova.js

```diff
@@ -91,85 +91,103 @@ CordovaCommand.prototype.run = function(options, callback) {
  */
 
 CordovaCommand.prototype.execute = function(options, callback) {
-
     var self = this;
 
     // in order to shell out to the cordova dependency, we must first find
-    // where the dependencies binary was installed.
-    //
-    // in npm < 3.x it is found in:
-    //   - `phonegap/node_modules/.bin/cordova`
-    //
-    // in npm >= 3.x it is either flattened or found in a local node_modules:
-    //   - `phonegap/node_modules/.bin/cordova`
-    //   - `/node_modules/.bin/cordova`
-    //     - where phonegap is found at `/node_modules/phonegap/`
-
-    // first try to find the cordova binary for npm < 3.x
-    var binPath = path.resolve(path.join(__dirname, '..', '..', 'node_modules', '.bin'));
-
-    // second try to find the cordova binary for npm >= 3.x
-    if (!fs.existsSync(path.join(binPath, 'cordova'))) {
-        binPath = path.resolve(path.join(__dirname, '..', '..', '..', '.bin'));
-    }
+    // the dependencies binary within the project
 
-    var command = options.cmd.split(' ')[0],           // 'cordova' process name (it may change one day)
-        cordovaCommand = path.join(binPath, command), // /path/to/node_modules/.bin/cordova
-        execOptions = { async: true, silent: true },
+    var binPath,
+        nodeModules,
+        e,
+        projectRoot,
         buffer = [];
+    
+    // import the module
+    // call it with process.cwd() and self.phonegap;
+    // handle error logging with promise
+    // save whether or not we have already run this process
+    return cordovaDependency.exec(undefined, self.phonegap)
```

Comment:
Even though undefined is being passed, the cordovaDepenency gets the correct project directory.